client/src/components/SimpleModal.js

```diff
@@ -0,0 +1,94 @@
+import React, { useContext } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import Modal from "@material-ui/core/Modal";
+import {
+  MenuItem,
+  Checkbox,
+  ListItemText,
+  FormControl,
+} from "@material-ui/core";
+
+import { MyContext } from "../Provider";
+import CustomButton from "./CustomButton";
+import API from "../utils/API";
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    position: "absolute",
+    width: 400,
+    backgroundColor: theme.palette.background.paper,
+    border: "2px solid #000",
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+  },
+}));
+
+export default function SimpleModal(props) {
+  let { menuShow, setMenuShow, item } = props;
+  const classes = useStyles();
+  const [topics, setTopics] = React.useState([...item.topics]);
+  const state = useContext(MyContext);
+ 
+  const addTopic = (topicId) => {
+    let updatedTopicArray = [...topics].map((n) => n.id).concat(topicId);
+    API.addTopic(item.id, { topics: updatedTopicArray }).then((result) => {
+      state.fetchPosts();
+    });
+  };
+
+  const Modalbody = (
+    <div style={ {
+      top: `50%`,
+      left: `50%`,
+      transform: `translate(-50%, -50%)`,
+  }} className={classes.paper}>
+      <FormControl>
+        {state.topics.map((topic) => {
+          if (topics.map((n) => n.title).indexOf(topic.title) > -1) {
```

Comment:
This is a bit confusing. You have `state.topics`, `topics`, `topic`, and `n`.
Try to find a better way to name these things.